daft/functions/ai/__init__.py

```diff
@@ -66,6 +66,8 @@ def embed_text(
     *,
     provider: str | Provider | None = None,
     model: str | None = None,
+    max_retries: int = 3,
+    on_error: str = "raise",
```

Comment:
We can do `Literal["raise"] | Literal["log"] | Literal["ignore"]` here for better typing.